projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_cursor.py

```diff
@@ -203,24 +257,46 @@ def close(self) -> None:
 
     def _recover_operation(self, exception, managed_operation):
         # TODO: configurable generic re-try.
+        available_plugin_implementation = self.__managed_database_connection and (
+            type(self.__managed_database_connection).db_connection_recover_operation
+            != IDatabaseManagedConnection.db_connection_recover_operation
+            and type(
+                self.__managed_database_connection
+            ).db_connection_decorate_operation
+            != IDatabaseManagedConnection.db_connection_decorate_operation
```

Comment:
Why do we check for decorate_operation in _recover_operation?

Reply:
because RecoveryCurser is created using the decorate operation:

https://github.com/vmware/versatile-data-kit/blob/774690666d948761befaedbf95e6ca2c08551356/projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_cursor.py#L212